libs/pipelines/imu_fusion_context.cpp

```
@@ -0,0 +1,591 @@
+  const float expected = freq * dT_s;
+  const float actual = static_cast<float>(preint.size());
+  // Allow up to ~2 samples of slack for phase/jitter (IMU timestamps are rarely periodic to within
+  // a fraction of the nominal period — see e.g. multi_rgbd indoor where sample gaps span 5–20 ms
+  // around a 10 ms nominal). Bound the slack by half the expected count so short intervals can't
+  // hide a genuine dropout (otherwise e.g. expected=2 with actual=0 would still report healthy).
+  constexpr float kJitterMarginSamples = 2.f;
+  const float effective_jitter = std::min(kJitterMarginSamples, expected * 0.5f);
+  const float missing = std::max(0.f, expected - actual - effective_jitter);
+  const float drop_ratio = missing / expected;
+  if (drop_ratio > 0.1f) {
+    TraceWarning("Lost IMU msgs: %d, Frame time delta = %f [s], drop ratio = %f [%]", static_cast<int>(missing + 0.5f),
```

Clang (14.0.6)

[warning] do not call c-style vararg functions

(cppcoreguidelines-pro-type-vararg)